base/abstractarray.jl

```diff
@@ -388,6 +420,47 @@ julia> last([1; 2; 3; 4])
 """
 last(a) = a[end]
 
+"""
+    last(predicate, coll)
+
+Get the last element of `coll` satisfying `predicate` wrapped in [`Some`](@ref).
+
+If no element of `coll` satisfies `predicate`, return `nothing`.
+
+!!! compat "Julia 1.6"
+    This method was added in Julia 1.6.
+    
+# Examples
+```jldoctest
+julia> last(<(5), 1:10)
+Some(4)
+
+julia> isnothing(last(isodd, 2:2:10))
+true
+
+julia> something(last(iseven, [5; 3; 4; 2; 6; 9]))
+6
+
+julia> something(last(>(10), 1:10), 0)
+0
+```
+"""
+function last(predicate, itr)
+    out = nothing
+    for x in itr
```

Comment:
I'd be inclined to not include this O(n) implementation  — I'd rather only support iterators that support `Iterators.Reverse` iteration.

Reply:
Agreed. We don't include an O(n) implementation for regular `last` either.